### PR TITLE
Fako/gitlab deploy

### PR DESCRIPTION
**Pull request by fako**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,37 +26,46 @@ To stay close to the production environment it works well to run the project in
...
-Similarly for the Elastic cluster you need to add ``POL_ELASTIC_SEARCH_HOST=127.0.0.1`` to the ``.env`` file
-or add ``127.0.0.1 elasticsearch`` to your hosts file.
-Finally add ``127.0.0.1 harvester`` to your hosts file to be able to visit links generated by the harvester
-in your browser and to allow the service running outside of its container to reach the harvester API.
+If you want to run the project outside of a container you'll need to add the following to your hosts file:
+```
+127.0.0.1 postgres
+127.0.0.1 elasticsearch
+127.0.0.1 harvester
+127.0.0.1 service
```
Comment (fako) [inline README.md:55]: @naftalivanderloon dit is nieuw, voeg dat lokaal toe

File: commands/services/service/fabric.py [hunk 2]
```diff
@@ -48,8 +48,7 @@ def create_snapshot(conn):
 
 
 @task(help={
-    "snapshot_name": "The file name of the snapshot you want to restore. Defaults to last updated snapshot",
-    "migrate": "Whether to apply some changes to the snapshot file to migrate from a pre-AWS format"
```
Comment (fako) [inline commands/services/service/fabric.py:52]: Dit fixt de fab problemen die jij zag @naftalivanderloon

File: harvester/requirements.txt [hunk 3]
```diff
@@ -14,8 +14,8 @@ django-x-robots-tag-middleware==1.3.3
 markupsafe==2.0.1
 
 # AWS
-awscli==1.18.195
-boto3==1.16.35
+awscli==1.24.7
+boto3==1.23.7
```
Comment (fako) [inline harvester/requirements.txt:18]: pip install -r requirements.txt && docker-compose up --build
@naftalivanderloon

File: README.md [hunk 4]
```diff
@@ -26,37 +26,46 @@ To stay close to the production environment it works well to run the project in
...
+```
+This way you can reach these containers outside of the container network through their names.
+This is important for many setup commands as well as the integration tests and running the service locally.
 
 To finish the general setup you can run these commands to build all containers:
 
 ```bash
+invoke sync-repository-state
```
Comment (fako) [inline README.md:63]: Ook nieuw. Zal je moeten uitvoeren voor een nieuwe build.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,168 @@
+# This file is a template, and might need editing before it works on your project.
```
Comment (naftalivanderloon) [inline .gitlab-ci.yml]: do we need this comment?

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+#
+# A pipeline is composed of independent jobs that run scripts, grouped into stages.
+# Stages run in sequential order, but jobs within stages run in parallel.
+#
+# For more information, see: https://docs.gitlab.com/ee/ci/yaml/index.html#stages
+
+image: ubuntu:22.04
+
+variables:
+  POSTGRES_USER: postgres
```
Comment (naftalivanderloon) [inline .gitlab-ci.yml:4]: For configurablility maybe use variables for all of these

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+integration-test-job:
+  stage: integration-test
+  services:
+    - name: $AWS_CONTAINER_REGISTRY/harvester:$CI_COMMIT_SHA
+      alias: harvester
+    - name: $AWS_CONTAINER_REGISTRY/search-portal:$CI_COMMIT_SHA
+      alias: service
+    - name: docker:dind
+    - name: postgres:12.2-alpine
```
Comment (naftalivanderloon) [inline .gitlab-ci.yml]: would it be possible to use a variable for the postgres version, so we only have to change it in one place when upgrading

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    - name: docker.elastic.co/elasticsearch/elasticsearch:7.9.3
```
Comment (naftalivanderloon) [inline .gitlab-ci.yml]: would it be possible to use a variable for the elasticsearch version, so we only have to change it in one place when upgrading

File: .gitlab-ci.yml [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    POL_ELASTIC_SEARCH_ENABLE_DECOMPOUND_ANALYZERS: 0
+  script:
+    - docker ps
+    - cypress install
+    - invoke srv.setup-postgres --host postgres
+    - invoke hrv.setup-postgres --host postgres
+    - invoke hrv.index-dataset-version localhost -d test
+    - curl http://harvester:8888/
+    - curl http://service:8000/health
+    # TODO: we need Cypress to use service:8080 instead of localhost:8000
```
Comment (naftalivanderloon) [inline .gitlab-ci.yml]: do we have a ticket to do this, so we wont forget?

File: README.md [hunk 10]
```diff
@@ -143,26 +152,21 @@ Deploy
...
-```bash
-invoke aws.push <target-project-name>
-```
-
-To see a list of all currently available images for a project you can run:
+Before deploying you'll want to decide on a version number.
+To see a list of all currently available images for a project and the versions they are tagged with you can run:
 
 ```bash
 invoke aws.print-available-images <target-project-name>
```
Comment (naftalivanderloon) [inline README.md:148]: maybe clarify a bit with an example, not sure what to use here <target-project-name> ( I think service and harvester)

**Review by naftalivanderloon — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 5 above)
Comment (fako) [reply .gitlab-ci.yml]: No, not really. Will delete it.

File: .gitlab-ci.yml (hunk 6 above)
Comment (fako) [reply .gitlab-ci.yml:4]: They already are variables. These are defined in the global variables block. I think variables specified during a pipeline invocation would overwrite these if you want them to.

File: .gitlab-ci.yml (hunk 9 above)
Comment (fako) [reply .gitlab-ci.yml]: https://www.pivotaltracker.com/story/show/182261850

File: .gitlab-ci.yml (hunk 6 above)
Comment (naftalivanderloon) [reply .gitlab-ci.yml:4]: I mean like POL_POSTGRES_USER: $POSTGRES_USER
My understanding is that if you define these in a .env file, that one will used
By specifying POSTGRES_USER: postgres the user name is now hardcoded, right?

**Review by naftalivanderloon — APPROVED** [review]

**Pull request merged**